src/dmtcp_restart.cpp

```diff
@@ -404,8 +404,42 @@ getMtcpArgs()
   return mtcpArgs;
 }
 
+vector<string>
+getMtcpEnvironment(RestoreTarget *restoreTarget)
```

Comment:
This is an example of highly non-local program design.  I see that you want this in combination with the new design of restart plugins.  But let's look for something better.  Why can't we simply take the entire environment from the global 'environ' environment variable, for example.

1. The purpose of this function seems to be to hardwire in certain environment variables, while removing all other environment variables.
2. So, when testing or debugging, if we call 'dmtcp_restart', we see one set of environment variables, and if we call 'mtcp_restart' in the 'mtcp' directory we see different environment variables.  This is confusing when extending 'mtcp_restart' in the mtcp directory and then testing it.
3. There is no harm in keeping the extraneous environment variables.  We are going to call 'mtcp_restart', which will restore the checkpoint image.  So, all extraneous environment variables will disappear, except those which 'mtcp_restart' treats specially (like 'DMTCP_RESTART_PAUSE'.
4. If you really want the reader to be aware of special-purpose environment variables like 'DTMCP_RESTART_PAUSE', then adding a _local_ comment near where the reader sees it (e.g., in the mtcp dir or in the restart_plugin dir of MANA) is a better option.

(While I like the benefits of C++, this is one place where I feel that C++ encourages a non-local design paradigm.  I feel that this is a misfeature of C++.
    Some examples are:  a C++ method might be defined in the .h or the .cpp file.  One should look in both places.  A namespace or class may sometimes be omitted.  Look elsewhere to discover the true signature (because this saves a small amount of typing that a good editor would be glad to insert automatically).  An operator may be what it looks like, or there may be a separate file redefining it.  There is no syntactic indicator.  (This is different from C macros, where macros are capitalized.)
    This non-local design happens at the most basic level of C++, even before we talk about inheritance. )

Anyway, enough complaints from me.  It's probably because this is late at night.  But I do want to avoid more non-local design.